miden/src/cli/compile_lib.rs

```diff
@@ -0,0 +1,36 @@
+use assembly::{LibraryNamespace, MaslLibrary, Version};
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "Compile Library", about = "Compiles a library")]
+pub struct CompileLibraryCmd {
+    /// Path to a directory containg the `.masm` files which are part of the library
+    #[structopt(parse(from_os_str))]
+    dir: PathBuf,
+    /// Name of the top-level namespace, e.g. `mylib`.
+    #[structopt()]
+    namespace: String,
+}
```

Comment:
I tried to reuse the `CompileCmd`, but `compile` is written to `compile` a whole program, which may use multiple libraries, each with their own namespace, i couldn't think of a simple/clear interface to do that.